components/director/pkg/persistence/sql_error_mapper.go

```diff
@@ -17,20 +19,27 @@ func MapSQLError(err error, resourceType resource.Type, format string, args ...i
 	}
 
 	if err == sql.ErrNoRows {
+		log.Errorf("SQL: no rows in result set for '%s' resource type", resourceType)
 		return apperrors.NewNotFoundErrorWithType(resourceType)
 	}
 
 	pgErr, ok := err.(*pq.Error)
 	if !ok {
-		return apperrors.InternalErrorFrom(err, format, args...)
+		log.Errorf("Error while casting to postgres error. Actual error: %s", err)
+		return apperrors.NewInternalError("Unexpected error while executing SQL query")
 	}
 
+	log.Errorf("SQL Error: %s. Caused by: %s. DETAILS: %s", fmt.Sprintf(format, args...), pgErr.Message, pgErr.Detail)
+
+	log.Debug("Checking Postgres error code...")
 	switch pgErr.Code {
```

Comment:
The two logs in the switch cases are redundant having the previous one outside of the switch giving all the necessary information